### PR TITLE
Multiple ssl profile general rewrite rule set

### PR DESCRIPTION
**Pull request by paulridout**
### Jira link (if applicable)
SDT-115 (https://tools.hmcts.net/jira/browse/SDT-115)

### Change description ###
Changed main.tf so that rewrite rules are dynamically created from the details specified in the configuration file passed to the module rather than being hard coded in the module.

This also incorporates the changes from #16.

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [no] Does this PR introduce a breaking change

### CODE REVIEW

File: example/config.yaml [hunk 1]
```diff
@@ -11,4 +11,23 @@ gateways:
         ssl_enabled: true
         use_public_ip: true
         health_path_override: "/health/liveness"
-
+        add_rewrite_rule: true
```
Comment (thomast1906) [inline example/config.yaml:14]: want to include all example changes? 
certificates & if any additional

File: variables.tf [hunk 2]
```diff
@@ -102,8 +102,4 @@ variable "exclusions" {
   default = []
 }
 
-variable "trusted_client_certificate_data" {
-  type        = string
-  default     = null
-  description = "This variable will allow to send trusted client certificate data as string"
-}
+variable "trusted_client_certificate_data" {}
```
Comment (thomast1906) [inline variables.tf:105]: Is this used/needed now?

Comment (paulridout) [reply variables.tf:105]: The trusted_client_certificate_data variable is still used.

The trusted_client_certificate_data variable is set by the code that calls this module to a map containing client certificate names and the certificate themselves.

The dynamic trusted_client_certificate block in main.tf takes the client certificate name specified in the rootca_certificate_name property and looks it up in the trusted_client_certificate_data map to get the certificate, which it then uses to set the data property of the block.

The code in main.tf that looks up the certificate uses a conditional expression.  This [returns false](https://github.com/hmcts/terraform-module-apim-application-gateway/blob/31a7a9ebce779f41ad597d3fc3f8b0b665f9776f/main.tf#L231) if no certificate exists with that name.  This probably isn't correct - perhaps it should return an empty string or a default certificate?

File: example/config.yaml (hunk 1 above)
Comment (paulridout) [reply example/config.yaml:17]: Added new settings and corrected position of some existing settings.

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**